examples/basic_auth.rs

```diff
@@ -0,0 +1,13 @@
+//#![deny(warnings)]
+use headers::{authorization::Basic, Authorization};
+use warp::Filter;
+
+#[tokio::main]
+async fn main() {
+    let routes = warp::auth::basic("Realm name").map(|auth_header: Authorization<Basic>| {
+        // TODO: some password lookups done in here
```

Comment:
It might be nice for the example to actually include a method for password lookups

Reply:
well anything serious needs to pass in some form of database, file or at least a hashmap dragging in dependencies. easiest form is a silly `if name == password` check, but not sure that makes sense.

Reply:
If the goal of an example is to illustrate how to use the filter, it would be good to provide something closer to a real world example instead of a todo comment. Using the `dir` example as a template, the project provides an actual path that is populated with files. When that example is run, it actually serves content, which helps to illustrate _how_ to use that filter.

Reply:
I'll add a mutex guarded Hashmap then, everything else needs 3rd party libs. Okay like that?

Reply:
You may want to make it a `Arc<tokio::sync::Mutex<HashMap<String, String>>>` since almost any implementation is going to need to do some kind of `async` work. The `tokio` dependency will already be there from using `tokio::main`

Reply:
I realised a lock-guard is not needed, since this is read-only. example should be fine now

Reply:
My suggestion of using the tokio mutex was mostly to have some async work to perform in the handler, the new example is quite a bit more illustrative though

Reply:
Sure I understand, but anything that needs async work (e.g. update of password db while server is running) is a way too big example. And showing use of a Mutex when lock-free is actually good enough is not a good example either ;)
however my example is broken, as it does not return a 401 on auth-error. but this is what my comment in the parent thread discusses, how to do this best 🤔